src/controllers/hid/hidcontroller.cpp

```diff
@@ -243,13 +248,38 @@ int HidController::close() {
 bool HidController::poll() {
     Trace hidRead("HidController poll");
 
-    int result = hid_read(m_pHidDevice, m_pPollData, sizeof(m_pPollData) / sizeof(m_pPollData[0]));
-    if (result == -1) {
-        return false;
-    } else if (result > 0) {
-        Trace process("HidController process packet");
-        QByteArray outData(reinterpret_cast<char*>(m_pPollData), result);
-        receive(outData, mixxx::Time::elapsed());
+    int result = 1;
+    // This loop risks becoming a high priority endless loop in case processing
+    // the mapping JS code takes longer than the controller polling rate.
+    // This could stall other low priority tasks.
+    // There is no safety net for this because it has not been demonstrated to be
+    // a problem in practice.
+    while (result > 0) {
+        // Rotate between two buffers so the memcmp below does not require deep copying to another buffer.
+        unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+        m_iPollingBufferIndex = (m_iPollingBufferIndex + 1) % kNumBuffers;
+        unsigned char* pCurrentBuffer = m_pPollData[m_iPollingBufferIndex];
+
+        result = hid_read(m_pHidDevice, pCurrentBuffer, kBufferSize);
+        if (result == -1) {
```

Comment:
I would use early returns and avoid all the nesting:
- result < 0 -> return false
- result == 0 -> return true
- ...DEBUG_ASSERT(result > 0) and process the received package

Reply:
Sure, this obviates the need for the hacky `int result = 1` too: d4afeb5